core/CronArchive.php

```diff
@@ -626,6 +633,27 @@ public function setSegmentsToForceFromSegmentIds($idSegments)
         $this->segmentsToForce = $segments;
     }
 
+
+    /**
+     * @param int[] $idSegments
+     */
+    public function setSegmentsToSkipFromSegmentIds($idSegments)
+    {
+        /** @var SegmentEditorModel $segmentEditorModel */
+        $segmentEditorModel = StaticContainer::get('Piwik\Plugins\SegmentEditor\Model');
+        $segments = $segmentEditorModel->getAllSegmentsAndIgnoreVisibility();
+
+        $segments = array_filter($segments, function ($segment) use ($idSegments) {
+            return (!in_array($segment['idsegment'], $idSegments));
```

Comment:
@mikkeschiren looks good so far. The only thing I'm wondering if we could also filter by idSite? 

eg 
```php
in_array($segment['idsegment'], $idSegments) && (empty($segment['enable_only_idsite']) || $idSite == $segment['enable_only_idsite'])
```

And then we could throw an exception if an `idSegments`  were specified, but none matches any. This way it wouldn't by accident set no segments to force, and the user would be aware a wrong ID was set.